modules/text/src/text_detector_swt.cpp

```diff
@@ -0,0 +1,812 @@
+        for (unsigned int i = 0; i < chains.size(); i++) {
+            if (!chains[i].merged) {
+                newchains.push_back(chains[i]);
+            }
+        }
+        chains = newchains;
+        std::stable_sort(chains.begin(), chains.end(), chainSortLength);
+    }
+
+    std::vector<ChainedComponent> newchains;
+    std::vector<std::vector<SWTPoint>> componentsPointsVector;
+    vector<Component> finalComponents;
+    finalComponents.reserve(components.size());
+    bool componentIncluded [components.size()] = {false};
+    for (unsigned int i = 0; i < chains.size(); i++) {
+        if (chains[i].componentIndices.size() >= 3) {
+            newchains.push_back(chains[i]);
+            int xmin,xmax,ymin,ymax;
+            xmin = 1000000;
+            ymin = 1000000;
+            xmax = 0;
+            ymax = 0;
+            for (unsigned int j = 0; j < chains[i].componentIndices.size(); j++) {
+                if (componentIncluded[chains[i].componentIndices[j]]) continue;
+                componentIncluded[chains[i].componentIndices[j]] = true;
+                Component acceptedComponent = components[chains[i].componentIndices[j]];
+                std::vector<SWTPoint> componentPoints;
+                for (unsigned int k = 0; k < acceptedComponent.points.size(); k++) {
+                    componentPoints.push_back(acceptedComponent.points[k]);
+                    xmin = min(xmin, acceptedComponent.points[k].x);
+                    ymin = min(ymin, acceptedComponent.points[k].y);
+                    xmax = max(xmax, acceptedComponent.points[k].x);
+                    ymax = max(ymax, acceptedComponent.points[k].y);
+                }
+                componentsPointsVector.push_back(componentPoints);
+            }
+            int wd = xmax - xmin;
+            int ht = ymax - ymin;
+            chainedTextRegions.push_back(Rect(xmin, ymin, wd, ht));
+        }
+    }
+    finalComponents = filterComponents(SWTImage, componentsPointsVector, true);
+    chains = newchains;
+    std::stable_sort(chains.begin(), chains.end(), chainSortLength);
+
+    Mat outTemp( output.size(), CV_32FC1 );
+    renderComponents(SWTImage, finalComponents, outTemp);
+    outTemp.convertTo(outTemp, CV_8UC1, 255.);
+    cvtColor (outTemp, output, COLOR_GRAY2RGB);
+    renderComponentBBs(finalComponents, output);
+    return getComponentBBs(finalComponents);
+}
+}
+
+CV_EXPORTS_W void detectTextSWT (InputArray input, CV_OUT std::vector<cv::Rect>& result, bool dark_on_light, OutputArray & draw /*=noArray()*/, OutputArray & chainBBs /*=noArray()*/) {
+    CV_CheckTypeEQ(input.type(), CV_8UC3, "");
+
+
+    // Convert to grayscale
+    Mat grayImage(input.size(), CV_8UC1);
+    cvtColor(input, grayImage, COLOR_RGB2GRAY);
+    // Create Canny Image
+    double threshold_low = 175;
+    double threshold_high = 320;
+    Mat canny_edge_image(input.size(), CV_8UC1);
+    Canny (grayImage, canny_edge_image, threshold_low, threshold_high, 3);
```

Comment:
BTW, There is `Canny()` overloads with custom pre-computed dx/dy gradients.

> Mat canny_edge_image(input.size(), CV_8UC1);

Avoid allocation of result matrices.